### PR TITLE
Adding integration test for user_agent_suffix

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Adds regression integration coverage to ensure CosmosClientBuilder::with_user_agent_suffix is observable on actual outgoing requests when running through the Cosmos in-memory emulator, by introducing a request-observation hook in the emulator transport.

**Changes:**

Add an in-memory emulator RequestObserver trait and plumb it through InMemoryEmulatorHttpClient so tests can inspect outgoing requests.
Add new in-memory-emulator integration tests that capture User-Agent headers and assert suffix propagation (plus a negative control).
Register the new test module in the emulator test suite.

### CODE REVIEW

File: sdk/cosmos/azure_data_cosmos/tests/in_memory_emulator_tests/user_agent.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    );
+
+    let with_suffix = snapshots
+        .iter()
+        .filter(|s| {
+            s.user_agent
+                .as_deref()
+                .is_some_and(|ua| ua.contains(SUFFIX))
+        })
+        .count();
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure_data_cosmos/tests/in_memory_emulator_tests/user_agent.rs]: Fixed

File: sdk/cosmos/azure_data_cosmos_driver/src/in_memory_emulator/client.rs [hunk 2]
```diff
@@ -52,6 +55,19 @@ impl InMemoryEmulatorHttpClient {
         Arc::clone(&self.store)
     }
 
+    /// Attaches a [`RequestObserver`] that is invoked for every request the
+    /// emulator handles, before the request is routed.
+    ///
+    /// Intended for tests that need to assert on outgoing request shape
+    /// (e.g. that the configured `User-Agent` suffix actually reaches the
+    /// wire). Without an observer the dispatch path pays no overhead.
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure_data_cosmos_driver/src/in_memory_emulator/client.rs:63]: Noise

File: sdk/cosmos/azure_data_cosmos/tests/in_memory_emulator_tests/user_agent.rs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+    for snap in &snapshots {
+        if let Some(ua) = snap.user_agent.as_deref() {
+            assert!(
+                !ua.contains(SUFFIX),
+                "request {:?} {} unexpectedly carried suffix {SUFFIX:?} in User-Agent {ua:?}",
+                snap.method,
+                snap.url,
+            );
+        }
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure_data_cosmos/tests/in_memory_emulator_tests/user_agent.rs]: Fixed

**Review by Pilchie — APPROVED** [review]
